### ISSUE TITLE
Feature Request: Ability to inject link tokens to be used to resolve link references

### ISSUE THREAD

**Issue by ndossougbete** [chjj/marked#805]
## Use case

I am using marked to process separate markdown files into a website, and because files move around I would like to have my links transformed into appropriate relative links. Since I can't really process links inside of the markdown file, I thought about using link references and feeding how to resolve them directly to marked. Something like:
### MD file

``` md
blah blah [blah][foo]
```
### JS file

``` js
marked.options  = {
  links: {
    foo: {href: 'bar'}
  }
}
```

This currently does not work, and it doesn't look like there is an easy way to achieve that. 
## Current workaround

``` js
var marked = require('marked');
marked.Lexer.lex = function(src, options) {
  var lexer = new marked.Lexer(options);
  lexer.tokens.links.bar = {href: 'bar'};  // I redefine the function to add this line
  return lexer.lex(src);
};

marked.lexer = marked.Lexer.lex;
```
## TL;DR

It would be great if the API exposed a way for the lexer to be prefilled with link references, or maybe inject them in the tokens before using them, something like that.

**Comment by Feder1co5oave** [comment]
You could always inject the link definitions at the end of the source, before lexing it?

**Comment by ndossougbete** [comment]
I'm using marked through [metalsmith-markdown](https://github.com/segmentio/metalsmith-markdown), modifying the lexer is an easier workaround for me. I would have to make a metalsmith plugin (or hook some other way into marked) to inject the link definitions at the end of all my markdown files and snippets otherwise.

**Comment by joshbruce** [comment]
Closing as out of scope for this library - not a framework - just a converter. See #956
